Sources/SourceKitBazelBSP/RequestHandlers/SKOptions/SKOptionsHandler.swift

```diff
@@ -86,4 +86,13 @@ final class SKOptionsHandler {
             workingDirectory: initializedConfig.rootUri
         )
     }
+
+    // MARK: - InvalidatedTargetObserver
+
+    func invalidate(targets: Set<AffectedTarget>) throws {
+        // Only clear cache if at least one file was created or deleted
+        if targets.contains(where: { $0.kind == .created || $0.kind == .deleted }) {
+            extractor.clearCache()
+        }
+    }
```

Comment:
I wonder if in this case we can't invalidate only those specific `targets`, as because of the previous logic we should already have the correct build graph. But I think this is fine as a start point, we can iterate later 👍